tests/Aspire.EndToEnd.Tests/ProjectInfo.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+
+namespace Aspire.EndToEnd.Tests;
+
+public sealed class ProjectInfo
+{
+    [JsonIgnore]
+    public HttpClient Client { get; set; } = default!;
+
+    public EndpointInfo[] Endpoints { get; set; } = default!;
+
+    /// <summary>
+    /// Sends a GET request to the specified resource and returns the response message.
+    /// </summary>
+    public Task<HttpResponseMessage> HttpGetAsync(string bindingName, string path, CancellationToken cancellationToken = default)
+    {
+        var allocatedEndpoint = Endpoints.Single(e => e.Name == bindingName);
+        var url = $"{allocatedEndpoint.Uri}{path}";
+
+        return Client.GetAsync(url, cancellationToken);
+    }
+
+    /// <summary>
+    /// Sends a GET request to the specified resource and returns the response body as a string.
+    /// </summary>
+    public Task<string> HttpGetStringAsync(string bindingName, string path, CancellationToken cancellationToken = default)
+    {
+        var allocatedEndpoint = Endpoints.Single(e => e.Name == bindingName);
+        var url = $"{allocatedEndpoint.Uri}{path}";
+
+        return Client.GetStringAsync(url, cancellationToken);
+    }
+
+    public async Task WaitForHealthyStatusAsync(string bindingName, CancellationToken cancellationToken = default)
+    {
+        while (true)
+        {
+            try
+            {
+                var status = await HttpGetStringAsync(bindingName, "/health", cancellationToken);
+                if (status == "Healthy")
+                {
+                    return;
+                }
+            }
+            catch
+            {
+                await Task.Delay(100, cancellationToken);
```

Comment:
Do I understand it correctly that `Client.GetStringAsync` should be following the Policy to wait up to 5 minutes, and thus this call should fail only then? If so, why do we need this retry here?

I did see on helix that `Client.GetStringAsync` would fail after 100secs, and then it would retry because of this loop.

Reply:
Oh, the retry is for the unhealthy case, but does that need a try/catch?

I had added a `CWL` in the catch here and saw:
```
WaitForHealthyStatusAsync failed: System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
 ---> System.TimeoutException: The operation was canceled.
 ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
 ---> System.IO.IOException: Unable to read data from the transport connection: Operation canceled.
 ---> System.Net.Sockets.SocketException (125): Operation canceled
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.InitialFillAsync(Boolean async)
   at System.Net.Http.HttpConnection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.<SendCoreAsync>g__Core|5_0(HttpRequestMessage request, Boolean useAsync, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Resilience.Internal.ResilienceHandler.<>c.<<SendAsync>b__2_0>d.MoveNext()
--- End of stack trace from previous location ---
   at Polly.Outcome`1.ThrowIfException()
   at Microsoft.Extensions.Http.Resilience.Internal.ResilienceHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendCoreAsync>g__Core|5_0(HttpRequestMessage request, Boolean useAsync, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)
   at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at Aspire.EndToEnd.Tests.ProjectInfo.WaitForHealthyStatusAsync(String bindingName, CancellationToken cancellationToken) in /workspaces/aspire/tests/Aspire.EndToEnd.Tests/ProjectInfo.cs:line 45
```

This shouldn't happen if the policy is being used, correct?

Just curious, nothing blocking.

Reply:
Turns out this is https://github.com/dotnet/extensions/issues/4770. This will be fixed when we get v8.2 of Microsoft.Extensions.Http.Resilience.

Reply:
https://github.com/dotnet/aspire/pull/2094 updates to a version that contains this fix.